File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    answer = 0
    left = 1
    right = max(times)*n
    
    while left<=right:
        mid = (left+right)//2
        cnt = 0
        
        for t in times:
            cnt += mid//t
            
            if cnt >= n:
                break
                
        if cnt>=n:
            right = mid-1
        else:
            left= mid+1
        
    return left

File: 프로그래머스/3/43238. 입국심사/입국심사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = 0;
    long long left = 0;
    long long right = *max_element(times.begin(), times.end());
    right *= n;
    while(left<=right){
        long long mid = (left+right)/2;
        long long cnt=0;
        for(auto t:times){
            cnt+=mid/t;
        }
        if(cnt<n){
            left=mid+1;
        }
        else{
            right=mid-1;
        }
    }
    return left;
}